Core/Src/main.c

```diff
@@ -18,10 +18,8 @@
 /* USER CODE END Header */
 /* Includes ------------------------------------------------------------------*/
 #include "main.h"
-
-/* Private includes ----------------------------------------------------------*/
-/* USER CODE BEGIN Includes */
 #include <stdio.h>
+#include <task1_ale.h>
```

Comment:
Usually angle brackets are interpreted from the compiler as system headers, while quotes are interpreted as local headers.
You can find more [here](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf) at section 6.10.2.
As good rule of thumb include with angle brackets if it's a library header, while use quotes for project headers.
This also helps with readability.

```suggestion
#include "task1_ale.h"
```

Also, usually we put include files in this order:
- project
- libraries
- framework
- system

So by this logic we put task headers before standard library headers.